### ISSUE TITLE
Slint + vulkano collaboration

### ISSUE THREAD

**Issue by burbokop** [slint-ui/slint#6805]
There is a rust example `opengl_texture` in the repo, showning how to collaborate with opengl. Is there a way to collaborate with vulkan though rust `vulkano` crate the same way as in this opengl example? If not, it would be good to have this ability because `vulkan` (`vulkano`) is more idiomatic and modern way to work with gpu then opengl. If yes, it would be good to add an official example.

**Comment by tronical** [comment]
Agreed. Besides the new API to select a renderer by features (where Vulkan need to be added), the big question is: how to represent the different typed needed in the GraphicsAPI enum? (Device, queue index, instance, etc)

**Comment by SuchAFuriousDeath** [comment]
Maybe you could create wrapper types around the types that slint uses internally and then have as_raw methods on each. 

The wrappers could also implement Rust conversion traits for converting into the types from the most common vulkan libraries like vulkano/ash and so on (Those could be hidden behind feature gates).

At a first glance, it seems that at least vulkano and ash allow constructing various things from raw handles:
https://docs.rs/vulkano/latest/vulkano/?search=from_handle
https://docs.rs/ash/latest/ash/index.html?search=from_raw

**Comment by tronical** [comment]
I think you're right, passing the raw handles (u64?) should work.

**Comment by SuchAFuriousDeath** [comment]
Somewhere usize might be more appropriate, not sure

**Comment by SuchAFuriousDeath** [comment]
@tronical  What exact values do we want to expose in the GraphicsAPI Vulkan variant?
I'm thinking of opening a PR - I currently got this to work with the renderer-skia-vulkan feature:

![image](https://github.com/user-attachments/assets/41ca2677-1a88-4d1f-a7a3-02c37eda885f)
Selecting not done yet, but I think I know how to do that.

**Comment by tronical** [comment]
Great :). Yeah, I think instance, physical device, logical device, queue, queue family index, get_proc_addr for instance and one for device. But I don't know how to best handle the drawable part....
